packages/babel-parser/src/plugins/flow.js

```diff
@@ -1134,12 +1151,12 @@ export default (superClass: Class<Parser>): Class<Parser> =>
     ): N.FlowQualifiedTypeIdentifier {
       startPos = startPos || this.state.start;
       startLoc = startLoc || this.state.startLoc;
-      let node = id || this.parseIdentifier();
+      let node = id || this.parseIdentifier(true);
```

Comment:
I think we should also replace by `flowParseRestrictedIdentifier` here because
```
interface I extends bool.member {}
```
will [throw](https://flow.org/try/#0JYOwLgpgTgZghgYwgAgJLIgD0iAJgZ2QCMB7EgGwDoBbCao6ZAbwF8g) `unexpected reserved type`, though the error message is not optimal.

Reply:
Updated with tests!